packages/calypso-products/src/is-multi-year-domain-product.ts

```diff
@@ -0,0 +1,9 @@
+import type { ResponseCartProduct } from '@automattic/shopping-cart';
+
+/**
+ * @param product Product to check.
+ * @returns boolean indicating whether the product is a multi-year domain product.
+ */
+export function isMultiYearDomainProduct( product: ResponseCartProduct ): boolean {
```

Comment:
Not blocking; With these small check functions, it's good to have some tests. E.g. it returns true if a ResponseCartProduct is passed in that should be a multi year domain product, but false otherwise, since there's logic in the checkout relying on it, and it will give us confidence in any future refactors. 

We can follow up with tests here.